### PR TITLE
Simplify smoketest

### PR DESCRIPTION
**Pull request by hackaugusto**
This PR:

1. Expands the block covered by the debug flag for smoketests
1. removed code that only made sense when udp was a transport
1. Passes some arguments explicitly instead of using a dictionary
1. flattens the logical depth of a smoketest.

Instead of:

- raiden.ui.cli:smoketest
- setup_testchain_and_raiden / setup_testchain / setup_raiden
- matrix_server_starter
- run_smoketest
- redirect_stdout
- smoketest_perform_tests

it does:

- raiden.ui.cli:smoketest
- redirect_stdout / setup_testchain / matrix_server_starter
- run_smoketest

Which should a bit easier to follow

### CODE REVIEW

File: raiden/tests/integration/cli/conftest.py [hunk 1]
```diff
@@ -37,8 +37,14 @@ def raiden_testchain(testchain_provider, cli_tests_contracts_version):
         matrix_server="auto",
         print_step=lambda x: None,
         contracts_version=cli_tests_contracts_version,
-        testchain_setup=testchain_provider,
+        eth_client=testchain_provider["eth_client"],
+        eth_rpc_endpoint=testchain_provider["eth_rpc_endpoint"],
+        web3=testchain_provider["web3"],
+        base_datadir=testchain_provider["base_datadir"],
+        keystore=testchain_provider["keystore"],
```
Comment (hackaugusto) [inline raiden/tests/integration/cli/conftest.py]: Instead of using the `testchain_provider` dictionary, passing arguments explicitly

File: raiden/tests/utils/smoketest.py [hunk 2]
```diff
@@ -218,27 +150,10 @@ def get_private_key(keystore):
     return accmgr.get_privkey(addresses[0], DEFAULT_PASSPHRASE)
 
 
-@contextmanager
-def setup_testchain_and_raiden(
```
Comment (hackaugusto) [inline raiden/tests/utils/smoketest.py]: This context manager was unnecessary. Now the `ui:smoketest` does this directly.

File: raiden/tests/utils/smoketest.py [hunk 3]
```diff
@@ -218,27 +150,10 @@ def get_private_key(keystore):
...
-    with testchain_manager as testchain:
-        yield setup_raiden(transport, matrix_server, print_step, contracts_version, testchain)
-
-
 @contextmanager
 def setup_testchain(
-    eth_client: EthClient, print_step: Callable, free_port_generator: Iterator[int]
+    eth_client: EthClient, free_port_generator: Iterable[Port]
 ) -> ContextManager[Dict[str, Any]]:
-    print_step("Starting Ethereum node")
```
Comment (hackaugusto) [inline raiden/tests/utils/smoketest.py]: Removing the `print_step` simplifies the `cli` tests

File: raiden/tests/utils/smoketest.py [hunk 4]
```diff
@@ -294,19 +209,47 @@ def setup_testchain(
         )
 
 
-def setup_raiden(transport, matrix_server, print_step, contracts_version, testchain_setup):
+@contextmanager
+def setup_matrix_for_smoketest(
```
Comment (hackaugusto) [inline raiden/tests/utils/smoketest.py]: This is just a wrapper to print the colored output

File: raiden/tests/utils/smoketest.py [hunk 5]
```diff
@@ -294,19 +209,47 @@ def setup_testchain(
...
+    from raiden.tests.utils.transport import matrix_server_starter
+
+    print_step("Starting Matrix transport")
+
+    with matrix_server_starter(free_port_generator=free_port_generator) as ctx:
+        yield ctx
+
+
+@contextmanager
+def setup_testchain_for_smoketest(
```
Comment (hackaugusto) [inline raiden/tests/utils/smoketest.py]: This is just a wrapper to print the colored output

File: raiden/tests/utils/smoketest.py [hunk 6]
```diff
@@ -373,104 +316,91 @@ def setup_raiden(transport, matrix_server, print_step, contracts_version, testch
...
     print_step("Starting Raiden")
 
-    config = deepcopy(App.DEFAULT_CONFIG)
-    args["config"] = config
-    # Should use basic routing in the smoke test for now
-    # TODO: If we ever utilize a PFS in the smoke test we
-    #     need to use the deployed service registry, register the
-    #     PFS service there and then change this argument.
-    args["routing_mode"] = RoutingMode.BASIC
-    args["one_to_n_contract_address"] = None
```
Comment (hackaugusto) [inline raiden/tests/utils/smoketest.py]: The `args` were also edited on `ui:smoketest`, now they are grouped.

File: raiden/tests/utils/smoketest.py [hunk 7]
```diff
@@ -373,104 +316,91 @@ def setup_raiden(transport, matrix_server, print_step, contracts_version, testch
...
-                logfile.seek(0)
-                append_report("Ethereum Node log output", logfile.read())
-    append_report("Raiden Node stdout", raiden_stdout.getvalue())
-    if success:
-        print_step(f"Smoketest successful")
-    else:
-        print_step(f"Smoketest had errors", error=True)
-    return success
+    app = None
+    api_server = None
```
Comment (hackaugusto) [inline raiden/tests/utils/smoketest.py]: `api_server` was not being `stop`ed and `get`ed, so errors there could be missed by the smoketest.

File: raiden/tests/utils/smoketest.py [hunk 8]
```diff
@@ -373,104 +316,91 @@ def setup_raiden(transport, matrix_server, print_step, contracts_version, testch
...
-    # TODO: If we ever utilize a PFS in the smoke test we
-    #     need to use the deployed service registry, register the
-    #     PFS service there and then change this argument.
-    args["routing_mode"] = RoutingMode.BASIC
-    args["one_to_n_contract_address"] = None
-
-    raiden_stdout = StringIO()
-    maybe_redirect_stdout = contextlib.redirect_stdout(raiden_stdout)
-    if debug:
-        maybe_redirect_stdout = contextlib.nullcontext()
```
Comment (hackaugusto) [inline raiden/tests/utils/smoketest.py]: This is not related to the test itself. I moved it out and grouped with the other context managers.

File: raiden/ui/cli.py [hunk 9]
```diff
@@ -491,23 +496,44 @@ def version(short):
...
+        raise RuntimeError(f"Invalid transport type '{transport}'")
 
     if report_path is None:
         report_file = mktemp(suffix=".log")
     else:
         report_file = report_path
+
+    enable_gevent_monitoring_signal()
+    make_requests_insecure()
+    urllib3.disable_warnings(InsecureRequestWarning)
```
Comment (hackaugusto) [inline raiden/ui/cli.py]: I tried to group the environment changes here

File: raiden/ui/cli.py [hunk 10]
```diff
@@ -523,13 +549,6 @@ def append_report(subject: str, data: Optional[AnyStr] = None):
     append_report("Raiden version", json.dumps(get_system_spec()))
     append_report("Raiden log")
 
-    step_count = 7
-    if ctx.parent.params["transport"] == "matrix":
```
Comment (hackaugusto) [inline raiden/ui/cli.py]: Now `matrix` is the only transport.

File: raiden/ui/cli.py [hunk 11]
```diff
@@ -541,65 +560,101 @@ def print_step(description: str, error: bool = False) -> None:
             file=stdout,
         )
 
-    print_step("Getting smoketest configuration")
-    contracts_version = environment_type_to_contracts_version(
-        ctx.parent.params["environment_type"]
-    )
+    contracts_version = environment_type_to_contracts_version(environment_type)
+
+    try:
```
Comment (hackaugusto) [inline raiden/ui/cli.py]: Made the try block broader to allow postmortens on the context managers

File: raiden/ui/cli.py [hunk 12]
```diff
@@ -541,65 +560,101 @@ def print_step(description: str, error: bool = False) -> None:
...
-            if option_.name in args.keys():
-                args[option_.name] = option_.process_value(ctx, args[option_.name])
-            else:
-                args[option_.name] = option_.default
-
-        port = next(free_port_generator)
-
-        args["api_address"] = "localhost:" + str(port)
-
-        if args["transport"] == "matrix":
```
Comment (hackaugusto) [inline raiden/ui/cli.py]: Now `matrix` is the only transport.

File: raiden/ui/cli.py [hunk 13]
```diff
@@ -541,65 +560,101 @@ def print_step(description: str, error: bool = False) -> None:
...
+                    contract_addresses=contract_addresses,
+                    token=token,
                 )
-                success = False
-        else:
-            # Shouldn't happen
-            raise RuntimeError(f"Invalid transport type '{args['transport']}'")
+            finally:
+                if ethereum_nodes:
+                    for node_executor in ethereum_nodes:
```
Comment (hackaugusto) [inline raiden/ui/cli.py]: now this handles all ethereum nodes.

File: raiden/tests/integration/cli/conftest.py [hunk 14]
```diff
@@ -10,41 +10,38 @@
...
-
-
 @pytest.fixture(scope="module")
 def cli_tests_contracts_version():
     return RED_EYES_CONTRACT_VERSION
 
 
 @pytest.fixture(scope="module")
-def raiden_testchain(testchain_provider, cli_tests_contracts_version):
+def raiden_testchain(blockchain_type, port_generator, cli_tests_contracts_version):
```
Comment (hackaugusto) [inline raiden/tests/integration/cli/conftest.py]: The fixture `testchain_provider` was only used here, so I removed it and added executed `setup_testchain` directly

Comment (ulope) [reply raiden/tests/integration/cli/conftest.py]: Not quite. `testchain_provider` was `session` scoped, this is `module`. That means we restart the eth node for every module now.

It's probably not a huge deal, but arguably we want to move to restarting less stuff for every test, not more.

Comment (hackaugusto) [reply raiden/tests/integration/cli/conftest.py]: Humm, good point, I missed that.
